### ISSUE TITLE
Clearer support for manipulating data segments

### ISSUE THREAD

**Issue by dvogel** [libbpf/libbpf-rs#572]
I would like to manipulate the initial value of entries in `.bss` and/or `.data` while loading my BPF program. Some of the plumbing appears available but it isn't clear how to put it together.

* I can iterate over the maps with [`OpenObject::maps_iter_mut`](https://docs.rs/libbpf-rs/latest/libbpf_rs/struct.OpenObject.html#method.maps_iter_mut) but the `.bss` section name seems to be prefixed with the first 8 characters of my program name.
* With a single variable in the `.bss` section I can use [`OpenMap::set_initial_value`](https://docs.rs/libbpf-rs/latest/libbpf_rs/struct.OpenMap.html#method.set_initial_value) to set the value. However with multiple variables it is unclear how to set these. I would like to avoid starting the loading program prior to setting these but `OpenMap` seems like my only interface to the map overlying the `.bss` section and it lacks an `update` method.
* Even if I could address separate entries in the map, it seems (from some `bpf-objdump` inspect) that the section is aligned to the size of the largest variable. In order to avoid having to hardcode that in userspace code, it seems [there is a `bpf_map__def(map)->value_size`](https://github.com/torvalds/linux/blob/8efd0d9c316af470377894a6a0f9ff63ce18c177/tools/testing/selftests/bpf/prog_tests/global_data_init.c#L23) value to read but I don't see it exposed in this library.

So in summary, it would be nice to have either a clear API for doing this along with an example or a clear indication that this approach to communicating the between userspace and the BPF code isn't supported by this library.

**Comment by danielocfb** [comment]
> Furthermore, as a rule of thumb we also try to make C objects available as a fall back to be used with `libbpf-sys`. This has been a recent development and so it hasn't been done throughout, and it's still missing for `OpenMap`, it seems. Will see if it makes sense to add.

This will be improved with https://github.com/libbpf/libbpf-rs/pull/581, if we decide to merge it.

**Comment by dvogel** [comment]
FWIW #581 would clarify this situation quite a bit for me. When I was looking through the documentation is was wondering whether the presence of [`Map::as_libbpf_bpf_map_ptr`](https://docs.rs/libbpf-rs/latest/libbpf_rs/struct.Map.html#method.as_libbpf_bpf_map_ptr) versus the lack of a similar method on `OpenMap` was an indication that I would introduce issues if I tried to reach below the surface and manipulate the map directly before a certain point. Having a `AsLibbpfObject` trait implemented more broadly would clarify that any conventional use of an object as it would be done with libbpf should be compatible.

**Comment by danielocfb** [comment]
> FWIW #581 would clarify this situation quite a bit for me.

Thanks for the feedback!
